src/webgpu/shader/validation/expression/call/builtin/trig.spec.ts

```diff
@@ -0,0 +1,151 @@
+export const description = `
+Validation tests for trigonometry builtins.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+/// An array of values ranging from -1 to 2
+const kMinusOneToTwo = [-1.0, -0.9, -0.1, 0.0, 0.1, 0.5, 0.9, 1.0, 1.1, 1.5, 1.9, 2.0] as const;
+
+/// The evaluation stages to test
+const kEvaluationStages = ['constant', 'override'] as const;
+
+/// All floating-point scalar and vector types to test
+const kFloatingPointTypes = [
+  '',
```

Comment:
IIUC this is the AbstractFloat case? Presumably this means we aren't testing vectors of AbstractFloats here, but maybe that's OK?

Reply:
Added abstract vector types to this